src/lib/steps.ts

```diff
@@ -6,6 +6,10 @@ type PrimitiveType = number | boolean | string | Date;
 type Templatable<T> = T | string;
 export type Reference = Pipeline | string;
 
+export function isNotDeReferenced(pipelineOrReference: Reference): pipelineOrReference is string {
```

Comment:
Maybe `isReference` would a simpler name that avoids the double negation

Reply:
https://github.com/ToucanToco/weaverbird/pull/654/commits/b5e00576c0ba6322cb27e9e757ea083c5f26ec0a